README.md

```diff
@@ -73,34 +104,210 @@ The system uses dedicated queues for managing different job phases:
+  - EventBridge for scheduling
+
+> 🚨 **Important Note**: SNOS requires the `get_storage_proof` RPC endpoint to function.
+> Currently, this endpoint is not implemented in Madara.
+>
+> 🚧 Until madara implements the `get_storage_proof` endpoint, you need to run Pathfinder alongside Madara:
+>
+> - Madara will run in sequencer mode
+> - Pathfinder will sync with Madara
+> - The orchestrator will use Pathfinder's RPC URL for SNOS and state update fetching
+>
+> This setup is temporary until either:
+>
+> 1. SNOS is adapted to work without the `get_storage_proof` endpoint, or
+> 2. The `get_storage_proof` endpoint is implemented in Madara
+
+## 🚀 Installation & Setup
 
-1. **Madara Node**
+### Building from Source
 
-   - Required for block processing
-   - Follow setup instructions at [Madara Documentation](https://github.com/madara-alliance/madara)
+1. **Install System Dependencies**
 
-2. **Prover Service**
+   ```bash
+   # Ubuntu/Debian
+   sudo apt-get update
+   sudo apt install build-essential openssl pkg-config libssl-dev
+   sudo apt install python3.9 python3.9-venv python3.9-distutils libgmp-dev python3.9-dev
 
-   - ATLANTIC must be running
+   # For macOS
+   brew install openssl pkg-config gmp python@3.9
+   ```
 
-3. **Infrastructure Dependencies**
-   - MongoDB for job management
-   - AWS services (or Localstack for local development):
-     - SQS for queues
-     - S3 for data storage
-     - SNS for alerts
-     - EventBridge for scheduling
+2. **Install Rust** (Cross-platform)
 
-## 🚀 Installation & Setup
+   ```bash
+   curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+   source ~/.bashrc  # Or source ~/.zshrc for macOS
+   ```
+
+3. **Clone Repository**
+
+   ```bash
+   git clone https://github.com/madara-alliance/madara-orchestrator.git
+   cd madara-orchestrator
+   git submodule update --init
+   ```
+
+4. **Build SNOS**
+
+   ```bash
+   make snos
+   ```
+
+5. **Build Project**
+
+   ```bash
+   cargo build --release
+   ```
+
+### Local Development Setup
+
+1. **Install Docker** (Cross-platform)
+   Follow the official installation guides:
+
+   - [Ubuntu Installation Guide](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository)
+   - [macOS Installation Guide](https://docs.docker.com/desktop/install/mac-install/)
+
+2. **Install Foundry** (Cross-platform)
+
+   ```bash
+   curl -L https://foundry.paradigm.xyz | bash
+   foundryup
+   ```
+
+3. **Start Local Services**
+
+   ```bash
+   # Start MongoDB
+   docker run -d -p 27017:27017 mongo
+
+   # Start Localstack
+   docker run -d -p 4566:4566 localstack/localstack@sha256:763947722c6c8d33d5fbf7e8d52b4bddec5be35274a0998fdc6176d733375314
+
+   # Start Anvil in a separate terminal
+   anvil --block-time 1
+   ```
+
+4. **Setup Mock Proving Service**
+
+   🚧 This setup is for development purposes only:
+
+   ```bash
+   # Start the mock prover service using Docker
+   docker run -d -p 6000:6000 ocdbytes/mock-prover:latest
```

Comment:
let's use a karnot/madara-alliance naemspace

Reply:
as discussed, deferring as of now